### DISCUSSION TITLE
Can MOOSE sample the values of nodal variable(s) at each quadrature point in the domain?

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#22100] [Q&A General] [answered]
I know the [NodalValueSampler](https://mooseframework.inl.gov/source/vectorpostprocessors/NodalValueSampler.html) can sample nodal variables at each node in the domain. I want to know if there is a vectorpostprocessor can sample nodal variables at each quadrature point in the domain? By the way, I find the [Sampler1DReal](https://mooseframework.inl.gov/modules/thermal_hydraulics/source/vectorpostprocessors/Sampler1DReal.html) in Thermal Hydraulics Module. Sampler1DReal can sample material properties at all quadrature points in mesh block, maybe I can modify the Sampler1DReal to obtain the nodal variables at each quadrature point? But I'm not sure if it is too hard.

Ethan

**Comment by GiudGiud** [comment, marked as answer]
Hello

Can you convert a nodal variable to a material property using a ParsedMaterial>? (I ve never done it)
Then you could just use the MaterialVPP to get it at quadrature points
https://mooseframework.inl.gov/source/vectorpostprocessors/MaterialVectorPostprocessor.html

Guillaume

**Reply by Ethan-xj** [reply]
That sounds clever! I will have a try!
